### ISSUE TITLE
W5500 Ethernet overlay with pCP

### ISSUE THREAD

**Issue by dasfdlinux** [piCorePlayer/pCP-Kernels#9]
I have a RaspberryPi Media Center board (see [here](https://github.com/sonocotta/raspberry-media-center)) that uses a W5500 ethernet module for networking with a RaspberryPi Zero (no wireless). When using the Raspberry Pi OS I can enable the W5500 device overlay by appending `dtoverlay=w5500` to `/boot/config.txt` which adds a network interface. However, piCorePlayer does not seem to recognize the W5500 using the same overlay.

If I flash pCP 9.2.0 to my SD card I can see `/boot/overlays/w5500.dtbo` exists, but booting with that image (and a config.txt modified as above) still shows no interfaces when I run `ifconfig -a` other than the loopback `lo`.

I am pretty new to pCP, is this a bug or am I missing something about enabling this interface?

**Comment by paul-1** [comment]
We do not include the driver for wiznet.

drivers/net/ethernet/wiznet/w5100.ko
drivers/net/ethernet/wiznet/w5100-wpi.ko

Looks like your box is using a pizero, if that is the case, the driver would be in this archive.
https://repo.picoreplayer.org/repo/14.x/armv6/releases/RPi/src/kernel/modules-6.1.77-pcpCore.tar.xz

**Comment by dasfdlinux** [comment]
Thank you Paul, that was helpful.

I downloaded those drivers and copied them to my boot partition, then booted up pCP and mounted the boot partition, and finally used `insmod` to load them. That added an `eth0` to my network configuration and I was able to set a static ip address and test out the connection successfully.

> $ mount /mnt/mmcblk0p1
> $ sudo insmod /mnt/mmcblk0p1/w5100.ko
> $ sudo insmod /mnt/mmcblk0p1/w5100-wpi.ko
> $ sudo ifconfig eth0 192.168.0.2 netmask 255.255.255.0 broadcast 192.168.0.255 up
> $ ifconfig eth0
> eth0  Link encap:Ethernet  HWaddr...
> $ ping 192.168.0.1
> ... 64 bytes from 192.168.0.1...

The static IP and ping were just for testing purposes, but how should I properly add the drivers to pCP so they are available on boot? I have been unsuccessfully searching through docs.picoreplayer.org and forums.slimdevices.com but I would appreciate some additional pointers.

**Comment by paul-1** [comment]
Create an extension.  https://docs.picoreplayer.org/information/picore_extensions/

**Comment by dasfdlinux** [comment]
I finally got this working. Under Step 3 of the linked instructions, it says 
> If you haven’t already loaded the squashfs-tools.tcz extension, do so now.

I was trying to do this via command line rather than through the gui, and I was running into issues loading and using the extension. Eventually I switched to loading the extension via picoreplayer's web gui, and that worked okay for me. I set these drivers to be loaded at:
`tc@pCP:~$ ls /lib/modules/6.1.77-pcpCore/kernel/drivers/net/ethernet/`
`w5100-spi.ko  w5100.ko`

I appreciate the help you gave me. Hopefully these breadcrumbs help the next person with a similar issue.
